tests/test_snippets.py

```diff
@@ -7,8 +7,8 @@
 from stumpy.snippets import _get_all_profiles
 
 test_data = [np.random.uniform(-1000, 1000, [64]).astype(np.float64)]
-s = [6, 7, 8]
-percentage = [0.7, 0.8, 0.9]
+s = [3, 6, 7, 8]
+percentage = [0.4, 0.7, 0.8, 0.9]
```

Comment:
Is this the version that has precision issues before?

Reply:
So, this version (i.e. `s==3` and `percentage=0.4`) exposed the bug and we fixed in this PR. However, we also faced the loss-of-precision as well in this version. And, we fixed that in this PR as well